### PR TITLE
[stable/grafana] add support for database credentials

### PR DESCRIPTION
**Pull request by mikeraimondi**
**What this PR does / why we need it**:
Allows database credentials to be stored in a configurable secret.

### CODE REVIEW

File: stable/grafana/templates/deployment.yaml [hunk 1]
```diff
@@ -123,6 +135,18 @@ spec:
             periodSeconds: 10
           resources:
 {{ toYaml .Values.resources | indent 12 }}
+    {{- if .Values.database.cloudSQL.enabled  }}
+        - name: {{ .Chart.Name }}-cloudsql-proxy
+          image: "{{ .Values.database.cloudSQL.image.repository }}:{{ .Values.database.cloudSQL.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
```
Comment (sameersbn) [inline stable/grafana/templates/deployment.yaml]: `cloudSQL.image` should define it's own image pull policy, i.e. `cloudSQL.image.pullPolicy`

File: stable/grafana/values.yaml [hunk 2]
```diff
@@ -180,3 +180,20 @@ ldap:
...
+    enabled: false
+    # image:
+    #   repository: gcr.io/cloudsql-docker/gce-proxy
+    #   tag: 1.11
+    #   pullPolicy: IfNotPresent
+    # instanceConnectionName: ""
+    # port: 5432
+    # # `existingSecret` is a reference to an existing secret containing the CloudSQL instance credentials
+    # # for Grafana in a key `credentials.json`.
+    # existingSecret: ""
```
Comment (sameersbn) [inline stable/grafana/values.yaml]: Please uncomment the default values, else users would need to define all the params when they want to enable this feature.

**Review by sameersbn — CHANGES_REQUESTED** [review]

File: stable/grafana/Chart.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 name: grafana
-version: 1.3.0
+version: 1.3.1
```
Comment (timstoop) [inline stable/grafana/Chart.yaml]: As this adds a new feature, please bump the minor version.

**Review by timstoop — CHANGES_REQUESTED** [review]
Can you please document the new options in the README as well?

**Comment by timstoop** [comment]
There have been merges to stable/grafana, could you merge from master and fix any conflicts? You probably need to look at the version in Chart.yaml again after that.

File: stable/grafana/templates/deployment.yaml [hunk 4]
```diff
@@ -122,6 +122,17 @@ spec:
                   name: {{ .Values.smtp.existingSecret }}
                   key: password
             {{- end }}
+            {{- if .Values.database.existingSecret }}
+            - name: GF_DATABASE_USER
```
Comment (jasongwartz) [inline stable/grafana/templates/deployment.yaml:126]: Couldn't this be implemented using the changes from https://github.com/kubernetes/charts/pull/5435?

Comment (unguiculus) [reply stable/grafana/templates/deployment.yaml:126]: Agreed. Closing this.

**Pull request closed without merging**